compatibility-tests/compile-fail/tests/ui/attribute-duplication.rs

```diff
@@ -0,0 +1,28 @@
+mod enum_duplication {
```

Comment:
Did you also want a module for structs? Otherwise, I think it’s ok to remove the module here.

Reply:
I think there should be a module for structs, but I didn't get struct duplication checks working - the code structure for structs wasn't laid out the way it was for enums, and I was having a harder time figuring out how to do that.  I wanted to leave this module so it'd be clearer that enums aren't the only things to check.

Reply:
> I didn't get struct duplication checks working

Can you expand a bit on what problems arose?

Reply:
It was just that the enum-handling code's structure made it obvious what to do, and the struct-handling code wasn't as obvious for me, and I wasn't comfortable enough yet with the code style.

Now that I look back at it, I think there may only be one line to change, where we match `Source::From(t, e)` -- if we already have `Some`, then instead of setting `transformation`, we'd add an error.  Does that seem right?

Reply:
(And if that's the case, the same thing applies to enums; it's another type of duplication I hadn't caught.)

Reply:
> if we already have `Some`, then instead of setting `transformation`, we'd add an error

That would be fine, but we have [similar cases elsewhere](https://github.com/shepmaster/snafu/blob/da2d4aced258bdc443c90d8b3de6b127912e2af0/snafu-derive/src/lib.rs#L415), where we "pop one and exactly one" value from a `Vec`. 

Could we change these cases to push to a `Vec` and then reuse / repurpose that logic:

```rust
fn only_one<T>(v: Vec<T>) -> Option<T> { ... }

let from = only_one(froms).ok_or_else(...)
```

Reply:
Not 100% sure how to combine that with cases of zero-or-one, or as a not-immediately-fatal error.

Reply:
I'll give this a shot.

Reply:
I came up with something I'm pretty happy with - `AtMostOne`, which replaces those `Vec`s and creates errors for attributes beyond the first.  It handles the struct case and cleans up the enum cases, removes the weird indexing reverse, and builds nicely on `SyntaxErrors`.  I hope you like it.

It didn't replace `one_field_error` which wasn't as good a fit, but if you have ideas for merging the two I could do that.  (I imagine it'd take a `Fn` for the error generation, or something similar, but it seemed like more overhead than it was worth to replace this one case, when the simpler thing replaced all the rest.)